### PR TITLE
Add a test to show that postgres timestamps are broken

### PR DESCRIPTION
**Pull request by rautenrieth-da**
This PR shows that the append-only index database has an issue if the default time zone is not UTC.

### CODE REVIEW

File: ledger/participant-integration-api/src/test/suite/scala/platform/store/backend/StorageBackendPostgresSpec.scala [hunk 1]
```diff
@@ -31,4 +35,46 @@ final class StorageBackendPostgresSpec
...
+      }
+      _ = TimeZone.setDefault(TimeZone.getTimeZone("Pacific/Kiritimati"))
+      r2 <- executeSql(backend.maximumLedgerTime(Set(cid)))
+      _ = TimeZone.setDefault(TimeZone.getTimeZone("Pacific/Apia"))
+      r3 <- executeSql(backend.maximumLedgerTime(Set(cid)))
+    } yield {
+      withClue("R1") { r1 shouldBe Success(Some(let)) }
+      withClue("R4") { r4 shouldBe Success(Some(let)) }
+      withClue("R5") { r5 shouldBe Success(Some(let)) }
+      withClue("R2") { r2 shouldBe Success(Some(let)) }
```
Comment (rautenrieth-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/store/backend/StorageBackendPostgresSpec.scala:76]: Spoiler: fails here

File: ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsTimestamps.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+  }
+
+  // Some JDBC operations depend on the JVM default time zone
+  private def withDefaultTimeZone[T](tz: String)(f: Connection => T)(connection: Connection): T = {
+    val previousDefaultTimeZone = TimeZone.getDefault
+    TimeZone.setDefault(TimeZone.getTimeZone(tz))
+    val result = f(connection)
+    TimeZone.setDefault(previousDefaultTimeZone)
+    result
```
Comment (SamirTalwar-DA) [inline ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsTimestamps.scala:78]: Cleanup in a `finally` block.

```suggestion
    try {
      f(connection)
    } finally {
      TimeZone.setDefault(previousDefaultTimeZone)
    }
```

**Review by SamirTalwar-DA — COMMENTED** [review]
Clever!

**Comment by rautenrieth-da** [comment]
These tests will be added as part of https://github.com/digital-asset/daml/pull/10769.

**Pull request closed without merging**